File: Algorithms/Searching/Linear Search/linear-search.rb
=begin
    Linear Search
=end

def LinearSearch(arr, k)
    i = 0
    while(i<arr.length)
        if arr[i] == k
           return i
        end
        i += 1
    end
    return -1
end

print LinearSearch([3, 5, 12, 56, 92, 123, 156, 190, 201, 222], 12)

File: Algorithms/Searching/Linear Search/linear-search.hs
{-
-- Linear Search
-}

linearSearch :: Eq a => [a] -> a -> Bool

linearSearch [] _ = False

linearSearch (x:xs) y = (x == y) || linearSearch xs y

main = do
  let arr = [2,4,3,1,4]
  print $ linearSearch arr 1
  print $ linearSearch arr 5
